bossroom.js

```diff
@@ -32,6 +32,14 @@ export const BossRoomBot = {
         return RGBot.getEntitiesOnTeam(tickInfo, 1).filter(entry => !entry.broken).sort((a,b) => MathFunctions.distanceSq(position, a.position) - MathFunctions.distanceSq(position, b.position)).find(() => true);
     },
 
+    getDoorSwitch: (tickInfo) => {
+        return RGBot.getEntitiesOfType(tickInfo, "FloorSwitch")[0];
```

Comment:
There are 2 door switches in the game, did you only add the sensor to the one in the first room ?  Otherwise this is non-deterministic whether it will work reliably as it may pick the switch behind the door.

Reply:
Yeah I only added it to the first one for this demonstration.